crates/settlement-clients/starknet/src/lib.rs

```diff
@@ -0,0 +1,158 @@
+pub mod config;
+pub mod conversion;
+
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use color_eyre::eyre::eyre;
+use color_eyre::Result;
+use lazy_static::lazy_static;
+use mockall::{automock, predicate::*};
+use starknet::accounts::ConnectedAccount;
+use starknet::providers::Provider;
+use starknet::{
+    accounts::{Account, Call, ExecutionEncoding, SingleOwnerAccount},
+    core::{
+        chain_id,
+        types::{BlockId, BlockTag, FieldElement, FunctionCall},
+        utils::get_selector_from_name,
+    },
+    providers::{jsonrpc::HttpTransport, JsonRpcClient},
+    signers::{LocalWallet, SigningKey},
+};
+
+use config::StarknetSettlementConfig;
+use conversion::{slice_slice_u8_to_vec_field, slice_u8_to_field};
+use settlement_client_interface::{parse_and_validate_block_order, SettlementClient, SettlementVerificationStatus};
+use utils::env_utils::get_env_var_or_panic;
+
+#[allow(unused)]
+pub struct StarknetSettlementClient {
+    pub account: SingleOwnerAccount<Arc<JsonRpcClient<HttpTransport>>, LocalWallet>,
+    pub core_contract_address: FieldElement,
+}
+
+pub const ENV_PUBLIC_KEY: &str = "STARKNET_PUBLIC_KEY";
+pub const ENV_PRIVATE_KEY: &str = "STARKNET_PRIVATE_KEY";
+
+// Assumed the contract called for settlement l ooks like:
+// https://github.com/keep-starknet-strange/piltover
+
+lazy_static! {
+    pub static ref CONTRACT_WRITE_UPDATE_STATE_SELECTOR: FieldElement =
+        get_selector_from_name("update_state").expect("Invalid update state selector");
+    pub static ref CONTRACT_READ_STATE_BLOCK_NUMBER: FieldElement =
+        get_selector_from_name("stateBlockNumber").expect("Invalid update state selector");
+}
+
+#[automock]
+#[async_trait]
+impl SettlementClient for StarknetSettlementClient {
```

Comment:
we've this in Zaun already

Reply:
Added a TODO about this - will be done after Zaun is updated to latest starknet_rs version